os/using-environment-variables-in-systemd-units.md

```diff
@@ -0,0 +1,113 @@
+# here we clean previously defined ETCD_DISCOVERY environment variable, we don't need it as we've already bootstrapped etcd cluster and ETCD_DISCOVERY conflicts with ETCD_INITIAL_CLUSTER environment variable
+Environment="ETCD_DISCOVERY="
+Environment="ETCD_NAME=node4"
+Environment="ETCD_INITIAL_CLUSTER=node1=http://10.0.1.10:2380,node2=http://10.0.1.11:2380,node3=http://10.0.1.12:2380,node4=http://10.0.1.13:2380"
+Environment="ETCD_INITIAL_CLUSTER_STATE=existing"
+```
+
+Run `sudo systemctl daemon-reload` and `sudo systemctl restart etcd2` to apply your changes. You will see that your proxy node became cluster member:
+
+```
+etcdserver: start member 9bf1b35fc7761a23 in cluster 36cce781cb4f1292
+```
+
+Once your proxy node became member node and `etcdctl cluster-health` shows healthy cluster, you can remove your temporarily drop-in `sudo rm /run/systemd/system/etcd2.service.d/99-restore.conf && sudo systemctl daemon-reload`.
+
+## EnvironmentFile directive
+
+EnvironmentFile similar to Environment directive but reads the environment variables from a text file. The text file should contain new-line-separated variable assignments. 
+
+It is impossible to use scripts in Environment directive. So you can not dynamically define Environment, i.e. this doesn't work `Environment=/usr/bin/curl http://example.com/something`. When you need to update your environment values dynamically you can combine systemd service unit and EnvironmentFile directive.
+
+For example in CoreOS `flanneld.service` unit file creates `/run/flannel_docker_opts.env` environment file which is used by `docker.service` unit to configure Docker use flannel interface. You can use similar example in your containers' unit files:
+
+```
+cat fleet_machines.service
+[Unit]
+Description=Generates /etc/fleet_machines.env file
+After=etcd2.service
+Requires=etcd2.service
+After=fleet.service
+Requires=fleet.service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/sh -c "/usr/bin/echo -n FLEET_MACHINES= > /etc/fleet_machines.env"
+ExecStart=/usr/bin/sh -c "/usr/bin/fleetctl list-machines -fields='ip,machine' -no-legend=true -full=true | tr '\t' '=' | tr '\n' ',' >> /etc/fleet_machines.env"
+```
+
+```
+cat container.service
+[Unit]
+Description=My test docker container
+After=docker.service
+Requires=docker.service
+After=fleet_machines.service
+Requires=fleet_machines.service
+
+[Service]
+EnvironmentFile=/etc/fleet_machines.env
+ExecStartPre=-/usr/bin/docker kill %p
+ExecStartPre=-/usr/bin/docker rm %p
+ExecStartPre=/usr/bin/docker pull ubuntu:latest
+ExecStart=/usr/bin/docker run --rm --name %p -e FLEET_MACHINES ubuntu:latest bash -c 'while true; do echo "$FLEET_MACHINES"; sleep 1; done'
+```
+
+## More systemd Examples
+
+For more systemd examples, check out these documents:
+
+[Customizing Docker][customizing-docker]
+[Customizing the SSH Daemon][customizing-sshd]
+[Using systemd Drop-In Units][drop-in]
+
+[drop-in]: {{site.baseurl}}/os/docs/latest/using-systemd-drop-in-units.html
```

Comment:
You can make all of these links work on Github (ie remove the `{{site.baseurl}}/os/docs/latest/`) since we rewrite them as part of our sync process